### PR TITLE
Add monitoring to Trino

### PR DESCRIPTION
**Pull request by rimolive**
This PR implements additional k8s objects to add prometheus monitoring capabilities in Trino. This change will require the new `quay.io/opendatahub/trino:361` image as it contains the prometheus library to export metrics.

## Testing instructions
* Deploy the trino/base manifest in a cluster
  * You can use the command `kustomize build trino/base | {oc|kubectl} apply -n <namespace> -f -`
* Check if no pods are in `CrashLookbakOff` or `Error` state
* Check if the trino-service Service exposes the ports 8080 and 8082 (the last one is the prometheus port)
* As an optional step, create a route for trino-service Service and access the hostname. You should see the prometheus metrics.

### CODE REVIEW

File: trino/base/trino-config-monitoring-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      # counts
+      - pattern: 'presto.execution<name=(.+)><>(.+TotalCount.*): (.*)'
+        name: 'presto_execution_$1_$2_total'
+        type: COUNTER
+      - pattern: 'presto.memory<type=(.*), name=(.*)><>(.+): (.*)'
+        name: 'presto_memory_$1_$2_$3'
+        type: GAUGE
+      - pattern: 'presto.failuredetector<name=HeartbeatFailureDetector><>ActiveCount: (.*)'
+        name: 'presto_heartbeatdetector_activecount'
+        type: GAUGE
```
Comment (anishasthana) [inline trino/base/trino-config-monitoring-secret.yaml:80]: Missing new line

File: trino/base/trino-config-secret.yaml [hunk 2]
```diff
@@ -33,3 +48,5 @@ stringData:
   node.properties: |-
     node.environment=$(trino_environment)
     node.data-dir=/tmp/data/trino
+
+
```
Comment (anishasthana) [inline trino/base/trino-config-secret.yaml:52]: Extra new line?

**Review by anishasthana — CHANGES_REQUESTED** [review]
Could we also update the monitoring stack that is is deployed (https://github.com/opendatahub-io/odh-manifests/tree/master/prometheus/operator/base) to have an additional pod or service monitor that would scrape trino metrics?

File: trino/base/trino-service.yaml [hunk 3]
```diff
@@ -7,9 +7,14 @@ metadata:
     role: trino-coordinator
 spec:
   ports:
-    - protocol: TCP
+    - name: http
```
Comment (anishasthana) [inline trino/base/trino-service.yaml:10]: Could we give this a better name than http?

**Review by anishasthana — COMMENTED** [review]
Changes lgtm!

**Review by LaVLaS — APPROVED** [review]
/lgtm

Assuming CI runs cleanly

**Pull request merged**